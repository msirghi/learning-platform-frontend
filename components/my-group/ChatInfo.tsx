import React from 'react';
import { useTranslation } from '../../i18n';
import { newsMock } from '../../mocks/newsMocks';
import styles from '../../styles/modules/MyGroup.module.scss';
import { NewsRow } from './news/NewsRow';

export const ChatInfo = () => {
  const { t } = useTranslation();

  return (
    <div className={styles.chatInfoContainer}>
      <div className={styles.header}>
        <span>{t('myGroup:groupNews')}</span>
      </div>

      <div className={styles.newsContent}>
        {[...newsMock].map(({ date, description, title, id }, idx) => {
          return (
            <>
              <NewsRow
                key={id}
                date={date}
                title={title}
                description={description}
                author={'Nicolae'}
              />
              {idx !== newsMock.length - 1 && <hr className={styles.separator} />}
            </>
          );
        })}
      </div>
    </div>
  );
};
