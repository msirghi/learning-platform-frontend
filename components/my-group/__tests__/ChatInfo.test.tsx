import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ChatInfo } from '../ChatInfo';

describe('ChatInfo component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ChatInfo />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
