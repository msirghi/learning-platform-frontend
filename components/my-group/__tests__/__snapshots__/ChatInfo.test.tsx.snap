// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInfo component should match the snapshot 1`] = `
<div
  className="chatInfoContainer"
>
  <div
    className="header"
  >
    <span>
      myGroup:groupNews
    </span>
  </div>
  <div
    className="newsContent"
  >
    <NewsRow
      author="Nicolae"
      date={2000-11-22T00:00:00.000Z}
      description="Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in."
      key="1"
      title="News 1"
    />
    <hr
      className="separator"
    />
    <NewsRow
      author="Nicolae"
      date={2000-11-22T00:00:00.000Z}
      description="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae."
      key="2"
      title="News 2"
    />
  </div>
</div>
`;
